### PR TITLE
[bitnami/schema-registry] Relax ssl options verifications in schema registry

### LINKED ISSUE #74907: Too strict validation of environment variables in the schema registry
### Name and Version

bitnami/schema-registry:7.7.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

I’m deploying Schema Registry on GKE with Managed Service for Apache Kafka on Google Cloud using the Bitnami Helm Chart. Connecting to Managed Kafka requires the `SASL_SSL` protocol. However, when I pass `SASL_SSL://bootstrap.kafka-managed-cluster.rest-of-url:9092` as the `SCHEMA_REGISTRY_KAFKA_BROKERS`, valid `SCHEMA_REGISTRY_KAFKA_SASL_USERS` and valid `SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDS` , I encounter the following validation error:

>In order to configure the TLS encryption for communication with Kafka brokers, you must mount your schema-registry.keystore.jks and schema-registry.truststore.jks certificates ...

The issue seems to lie in this code block:

https://github.com/bitnami/containers/blob/e835fc0b86aaedf404ee9cb8f40e83228d9450d1/bitnami/schema-registry/7.7/debian-12/rootfs/opt/bitnami/scripts/libschemaregistry.sh#L147-L156

This validation is a false positive, as SASL_SSL does not require a keystore or truststore - it only needs the `SCHEMA_REGISTRY_KAFKA_SASL_USERS` and `SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDS` variables, which I’ve already set. This validation needs to be changed.

### What is the expected behavior?

Schema registry should connect to external Kafka using SASL_SSL as security protocol.

### What do you see instead?

>In order to configure the TLS encryption for communication with Kafka brokers, you must mount your schema-registry.keystore.jks and schema-registry.truststore.jks certificates ...

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by michalmisiewicz**
### Description of the change

The Schema Registry `libschemaregistry.sh` script enforces too strict validation of SSL environment variables, which prevents integration with [Google Cloud Managed Service for Apache Kafka](https://cloud.google.com/products/managed-service-for-apache-kafka?hl=en). This service requires SASL_SSL authentication without the use of a keystore or truststore. Details of the required configurations can be found [here](https://cloud.google.com/managed-service-for-apache-kafka/docs/authentication-kafka#sasl-plain).

In this PR, I relaxed the SSL validation rules to enable seamless integration with Managed Kafka. 

I removed also the verification for `SCHEMA_REGISTRY_KAFKA_SASL_USERS` and `SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDS`, as these settings are not required for all `sasl.mechanism` configurations. Example of configuration for `sasl.mechanism= OAUTHBEARER` can be found [here](https://github.com/googleapis/managedkafka?tab=readme-ov-file#kafka-java-auth-client-handler).

### Benefits

Schema registry image can connect to Google Cloud Managed Service for Apache Kafka. 

### Applicable issues

- fixes #74907 

### Additional information

I tested the update with Google Cloud Managed Service for Apache Kafka and successfully established a secure connection with Kafka.

### CODE REVIEW

**Review by juan131 — APPROVED** [review]
LGTM! Waiting for CI to be successful

File: bitnami/schema-registry/7.4/debian-12/rootfs/opt/bitnami/scripts/libschemaregistry.sh [hunk 1]
```diff
@@ -144,16 +144,11 @@ schema_registry_validate() {
 
     if [[ -n "$SCHEMA_REGISTRY_KAFKA_BROKERS" ]]; then
         if brokers_auth_protocol="$(schema_registry_brokers_auth_protocol)"; then
-            if [[ "$brokers_auth_protocol" =~ SSL ]]; then
+            if [[ "$brokers_auth_protocol" = SSL ]]; then
```
Comment (juan131) [inline bitnami/schema-registry/7.4/debian-12/rootfs/opt/bitnami/scripts/libschemaregistry.sh:147]: What about keeping the current check but replacing `print_validation_error` with a simple warning:
```
warn "In order to configure the TLS encryption for communication with Kafka brokers, most auth protocols require mounting your schema-registry.keystore.jks and schema-registry.truststore.jks certificates to the ${SCHEMA_REGISTRY_CERTS_DIR} directory."
```

File: bitnami/schema-registry/7.4/debian-12/rootfs/opt/bitnami/scripts/libschemaregistry.sh [hunk 2]
```diff
@@ -144,16 +144,11 @@ schema_registry_validate() {
...
                 if [[ ! -f ${SCHEMA_REGISTRY_CERTS_DIR}/schema-registry.keystore.jks ]] || [[ ! -f ${SCHEMA_REGISTRY_CERTS_DIR}/schema-registry.truststore.jks ]]; then
                     print_validation_error "In order to configure the TLS encryption for communication with Kafka brokers, you must mount your schema-registry.keystore.jks and schema-registry.truststore.jks certificates to the ${SCHEMA_REGISTRY_CERTS_DIR} directory."
                 fi
             fi
-            if [[ "$brokers_auth_protocol" =~ SASL ]]; then
-                if [[ -z "$SCHEMA_REGISTRY_KAFKA_SASL_USERS" ]] || [[ -z "$SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDS" ]]; then
-                    print_validation_error "In order to configure SASL authentication for Kafka, you must provide the SASL credentials. Set the environment variables SCHEMA_REGISTRY_KAFKA_SASL_USERS and SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDs"
-                fi
-            fi
```
Comment (juan131) [inline bitnami/schema-registry/7.4/debian-12/rootfs/opt/bitnami/scripts/libschemaregistry.sh:156]: Sth similar here, we can replace the current error with a warning.

**Review by juan131 — CHANGES_REQUESTED** [review]
After rethinking about this, I think we could preserve validations but using warnings instead of errors given they're applicable for most common use cases. Full example:

```bash
            if [[ "$brokers_auth_protocol" =~ SSL ]]; then
                if [[ ! -f ${SCHEMA_REGISTRY_CERTS_DIR}/schema-registry.keystore.jks ]] || [[ ! -f ${SCHEMA_REGISTRY_CERTS_DIR}/schema-registry.truststore.jks ]]; then
                    warn "In order to configure the TLS encryption for communication with Kafka brokers, most auth protocols require mounting your schema-registry.keystore.jks and schema-registry.truststore.jks certificates to the ${SCHEMA_REGISTRY_CERTS_DIR} directory."
                fi
            fi
            if [[ "$brokers_auth_protocol" =~ SASL ]]; then
                if [[ -z "$SCHEMA_REGISTRY_KAFKA_SASL_USERS" ]] || [[ -z "$SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDS" ]]; then
                    warn "In order to configure SASL authentication for Kafka, most auth protocols require providing the SASL credentials. Set the environment variables SCHEMA_REGISTRY_KAFKA_SASL_USERS and SCHEMA_REGISTRY_KAFKA_SASL_PASSWORDS if your auth protocol requires it."
                fi
            fi
```

**Comment by michalmisiewicz** [comment]
@juan131 I've just updated the code

**Comment by michalmisiewicz** [comment]
I’ve just tested the latest version with Managed Kafka, and it works perfectly

**Review by juan131 — APPROVED** [review]
Thanks so much for applying the suggestions!

**Comment by juan131** [comment]
Triggering a new container release, it should be available in the next few hours

**Pull request merged**